### PR TITLE
Add key feature doc

### PR DESCRIPTION
**Pull request by jwcesign**
Signed-off-by: jwcesign <jiangwei115@huawei.com>

**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:
Add key feature doc

**Which issue(s) this PR fixes**:
none

**Special notes for your reviewer**:
none

### CODE REVIEW

**Review by RainbowMango — COMMENTED** [review]
/assign @Poor12

File: docs/key-features/features.md [hunk 1]
```diff
@@ -2,4 +2,89 @@
 title: Key Features
 ---
 
-TBD
+## Cross-cloud multi-cluster multi-mode management
```
Comment (Poor12) [inline docs/key-features/features.md:5]: How about add a feature list before introducing the feature?

File: docs/key-features/features.md [hunk 2]
```diff
@@ -2,4 +2,89 @@
...
+  * ReplicasScheduling: Replication mode and split mode for instanced workloads.
+
+* Differential configuration (OverridePolicy)
+  * ImageOverrider: Differentiated configuration of mirrors.
+  * ArgsOverrider: Differentiated configuration of execution parameters.
+  * CommandOverrider: Differentiated configuration for execution commands.
+  * PlainText: Customized Differentiation Configuration.
+
+The overall rescheduling process is shown in the following figure:  
+![overall-relationship.png](../resources/key-features/overall-rescheduling.png)
```
Comment (Poor12) [inline docs/key-features/features.md:36]: Add a blank line between words and  picture? I think some places look overcrowded on the website.

**Review by Poor12 — COMMENTED** [review]
I think we should add some links from key features to userguide. So users can easily jump over.

File: docs/key-features/features.md [hunk 3]
```diff
@@ -2,4 +2,89 @@
...
+
+## Separating the concerns of different roles
+Users can access all member clusters with unified authentication:
+![overall-relationship.png](../resources/key-features/unified-access.png)
+
+* Aggregate API unified access entry.
+* Access control is consistent with member clusters.
+
+Users also can defined global resource quota via `FederatedResourceQuota`:  
+![overall-relationship.png](../resources/key-features/unified-access.png)
```
Comment (Poor12) [inline docs/key-features/features.md:76]: This pic seems misplaced.

Comment (jwcesign) [reply docs/key-features/features.md]: fixed

File: docs/key-features/features.md (hunk 2 above)
Comment (jwcesign) [reply docs/key-features/features.md:45]: fixed

File: docs/key-features/features.md (hunk 1 above)
Comment (jwcesign) [reply docs/key-features/features.md:5]: I move the feature list first

File: docs/key-features/features.md [hunk 4]
```diff
@@ -2,4 +2,109 @@
...
+
+![overall-relationship.png](../resources/key-features/unified-search.png)
+
+## Separating the concerns of different roles
+karmada supports:
+* [Unified authentication](https://karmada.io/docs/userguide/roleseparation/unifiedAuth):
+  * Aggregate API unified access entry.
+  * Access control is consistent with member clusters.
+* Unified resource quota(`FederatedResourceQuota`):
+  * Globally configure the ResourceQuota of each member cluster.
```
Comment (Poor12) [inline docs/key-features/features.md:86]: configures

File: docs/key-features/features.md [hunk 5]
```diff
@@ -2,4 +2,109 @@
...
+Users also can defined global resource quota via `FederatedResourceQuota`:  
+
+![overall-relationship.png](../resources/key-features/unified-resourcequota.png)
+
+## Cross-cluster service governance
+karmada supports:
+* [Multi-cluster service discovery](https://karmada.io/docs/userguide/service/multi-cluster-service):
+  * With ServiceExport and ServiceImport, achieving cross-cluster service discovery.
+* [Multi-cluster network support](https://karmada.io/docs/userguide/network/working-with-submariner):
+  * Use `Submarine` to open up the container network between clusters.
```
Comment (Poor12) [inline docs/key-features/features.md:105]: Use Submariner

**Review by Poor12 — COMMENTED** [review]
Add it in zh/docs, too.

File: docs/key-features/features.md [hunk 6]
```diff
@@ -2,4 +2,109 @@
...
+
+If one cluster does not have enough resource to accommodate their pods, Karamda will reschedule the pods. The overall rescheduling process is shown in the following figure:  
+
+![overall-relationship.png](../resources/key-features/overall-rescheduling.png)
+
+
+## Cross-cluster failover of applications
+Karmada supports:
+* [Cluster failover](https://karmada.io/docs/userguide/failover/):
+  * Karmada supports users to set distribution policies, and automatically migrates the faulty cluster instances in a centralized or decentralized manner after a cluster failure.
```
[the remark this answers is not in the document]
Comment (jwcesign) [reply docs/key-features/features.md:51]: How about change to cluster workloads? or directly replicas?

[the remark this answers is not in the document]
Comment (jwcesign) [reply docs/key-features/features.md]: change to replicas

File: docs/key-features/features.md (hunk 4 above)
Comment (jwcesign) [reply docs/key-features/features.md]: fixed

File: docs/key-features/features.md (hunk 5 above)
Comment (jwcesign) [reply docs/key-features/features.md]: fixed

File: docs/key-features/features.md [hunk 7]
```diff
@@ -2,4 +2,109 @@
...
+* Cluster taint settings:
+  * When the user sets a taint for the cluster and the resource distribution strategy cannot tolerate the taint, Karmada will also automatically trigger the migration of the cluster instance.
+* Uninterrupted service:
+  * During the instance migration process, Karmada can ensure that the service instance does not drop to zero, thereby ensuring that the service will not be interrupted.
+
+Karmada supports failover for clusters, one cluster failture will cause failover of instance as follows:
```
[the remark this answers is not in the document]
Comment (jwcesign) [reply docs/key-features/features.md]: fixed

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**